src/authorization/AuthorizationsDefinition.ts

```diff
@@ -1167,11 +1157,62 @@ export const AUTHORIZATION_DEFINITION: AuthorizationDefinition = {
           'user.name', 'user.firstName', 'actionOnUser.name', 'actionOnUser.firstName', 'hasDetailedMessages', 'detailedMessages'
         ]
       },
-      { resource: Entity.TOKEN, action: Action.LIST },
       {
-        resource: Entity.TOKEN,
-        action: [Action.CREATE, Action.READ, Action.UPDATE, Action.DELETE],
-        args: { 'sites': '$.site' }
+        resource: Entity.REGISTRATION_TOKEN, action: Action.LIST,
+        condition: {
+          Fn: 'custom:dynamicAuthorizations',
+          args: {
+            asserts: [],
+            filters: ['SitesAdmin'],
+            metadata: {
+              siteAreaID: {
+                mandatory: true,
+              }
+            }
+          }
+        },
+        attributes: [
+          'id', 'status', 'description', 'createdOn', 'lastChangedOn', 'expirationDate', 'revocationDate',
+          'siteAreaID', 'siteArea.name', 'createdBy.name', 'createdBy.firstName', 'lastChangedBy.name', 'lastChangedBy.firstName'
+        ]
+      },
+      {
+        resource: Entity.REGISTRATION_TOKEN, action: [Action.READ],
+        condition: {
+          Fn: 'custom:dynamicAuthorizations',
+          args: {
+            asserts: [],
+            filters: ['SitesAdmin'],
+          }
+        },
+        attributes: [
+          'id', 'status', 'description', 'createdOn', 'lastChangedOn', 'expirationDate', 'revocationDate',
+          'siteAreaID', 'siteArea.name', 'createdBy.name', 'createdBy.firstName', 'lastChangedBy.name', 'lastChangedBy.firstName'
+        ]
+      },
+      {
+        resource: Entity.REGISTRATION_TOKEN, action: [Action.CREATE, Action.UPDATE],
+        condition: {
+          Fn: 'custom:dynamicAuthorizations',
+          args: {
+            asserts: ['SiteAreaMandatory'],
```

Comment:
Site Admin can create registration tokens of site different from site in which his not site Admin via Postman.

Reply:
Good catch!
Feel free to provide a fix ;-)

Reply:
Normally, the code below should prevent it:
![2021-12-14_15-15-27](https://user-images.githubusercontent.com/25266945/146015195-85928112-2707-4fcc-bed7-ea2a0a67d7a2.png)

Reply:
@LucasBrazi06  I think the issue comes from the Action.READ which is not restrictive enough, since a **Site Admin** can read / see all site areas associated to his sites / organisation. 
And as a consequence if we provide in the Postman request a site area that is in his site / organisation but NOT in his assigned site, the filter in checkAndGetSiteAreaAuthorization won't work and the call will succeed !
To confirm my theory I used Action.READ_ASSETS_FROM_SITE_AREA and after the modification the behavior works as expected since the action has a SiteAdmin filter